js/ui/tab_cookies.js

```diff
@@ -156,6 +156,52 @@
           }
         });
         break;
+      case "whitelist-cookie-domain":
+        var domain = getParentData(element, "data-access");
+        var url = getUrl(domain, "", true);
+        getStorage("cookieWhitelist", function(data) 
+        {
+          let whitelist = data["cookieWhitelist"]
+          if (!whitelist.propertyIsEnumerable(domain)) 
+          {
+            whitelist[domain] = []
+          }
+          if (!whitelist[domain].includes("")) 
```

Comment:
Question: Not sure what is the intention here of checking for an empty item in the array ? 

I assume what you want to do here is to check whether user wish to whitelist whole domain or a specific cookie for a domain:

Suggestion: For that reason I would think of a current data structure for storing "cookieWhitelist":
```
{
  cookieWhitelist: {
    "amazon.com": 
    {
      domainWhitelist: false,
      cookies: ["aws-userInfo", "session"...]
    },
    "login.google.com":
    {
      domainWhitelist: true,
      cookies: []
    }
  }
  ...
} 
```
```
const domain = getParentData(element, "data-access");

getStorage("cookieWhitelist", function(cookieWhitelist)
{
    if (domain in cookieWhitelist && cookieWhitelist[domain].domainWhitelist)
    {
      cookieWhitelist[domain].domainWhitelist = false;
    }
    else
    {
      // set domainWhitelist = true;
    }
}
```

This way it would be possible to more easily manage the whitelisted cookies and domains and not loose the states when switching between whitelisting enabled and disabled states.

(Unless I'm missing something)

Reply:
Removing and adding items into cookies array for specific domain I think should be handled in `whitelist-sublist-cookie`.

Reply:
> Question: Not sure what is the intention here of checking for an empty item in the array ?
yup, this is correct, as cookie names can't be empty, it's easy to check against an empty string as the "domain"
though that's not clear when reading the code
the suggested structure is definitely better

Reply:
> I assume what you want to do here is to check whether user wish to whitelist whole domain or a specific cookie for a domain:

not completely correct
it's simply to check if a domain was already whitelisted
since an empty string is stored if a domain is whitelisted (since cookie names can't be empty, it's easy to store just an empty string as a domain whitelist)
this is not related to whitelisting specific cookies for a domain
though it could be more clear

the logic for whitelisting specific cookies is in whitelist-sublist-cookie

Reply:
> I assume what you want to do here is to check whether user wish to whitelist whole domain or a specific cookie for a domain:
>  > it's simply to check if a domain was already whitelisted

I forgot to edit my message, after finalizing the suggestion. You are correct, this part of the code is only related to Domain whitelist.

> since cookie names can't be empty, it's easy to store just an empty string as a domain whitelist

I see.

> though it could be more clear

I think if you would keep something similar to a structure mentioned above, it would be more readable and intuitive.